File: frontend/src/api/authApi.js
import apiClient from './axios'

export function login(email, password) {
  return apiClient.post('/api/auth/login', { email, password }).then((res) => res.data)
}

export function register({ fullName, email, password }) {
  return apiClient
    .post('/api/auth/register', { fullName, email, password, role: 'FARMER' })
    .then((res) => res.data)
}

File: mobile/lib/api/auth_api.dart
import '../core/api_client.dart';
import '../core/auth_storage.dart';
import '../core/json.dart';
import '../models/auth_session.dart';
import '../models/user.dart';

class AuthApi {
  static Future<AuthSession> login(String email, String password) async {
    final data = await ApiClient.instance.post<dynamic>(
      '/api/auth/login',
      data: {'email': email, 'password': password},
    );
    final session = AuthSession.fromJson(asMap(data));
    await AuthStorage.instance.save(session);
    return session;
  }

  static Future<User> register({
    required String fullName,
    required String email,
    required String password,
  }) async {
    final data = await ApiClient.instance.post<dynamic>(
      '/api/auth/register',
      data: {'fullName': fullName, 'email': email, 'password': password, 'role': 'FARMER'},
    );
    return User.fromJson(asMap(data));
  }

  static Future<void> logout() => AuthStorage.instance.clear();
}
